pyinfra/operations/git.py

```diff
@@ -149,18 +149,23 @@ def repo(
 @operation()
 def worktree(
     worktree,
-    repo=None, branch=None, create_branch=False, detached=False,
+    repo=None, detached=False, new_branch=None, commitish=None,
+    pull=True, rebase=False, from_remote_branch=None,
     present=True, assume_repo_exists=False, force=False,
     user=None, group=None, state=None, host=None,
 ):
     '''
     Manage git worktrees.
 
-    + repo: git main repository directory
     + worktree: git working tree directory
-    + branch: branch to use for the working tree
-    + create_branch: the branch already exist or should be created
+    + repo: git main repository directory
     + detached: create a working tree with a detached HEAD
+    + new_branch: local branch name created at the same time than the worktree
```

Comment:
To avoid breaking changes here we need to keep the old arguments valid (deprecated is fine) - I think `branch` can be left as is (not rename to `new_branch`).

For `create_branch` - is it possible to create a worktree using an existing branch? Ie calling with `branch=x, create_branch=False`)?

Reply:
> To avoid breaking changes here we need to keep the old arguments valid (deprecated is fine) - I think `branch` can be left as is (not rename to `new_branch`).

Yes, true! To avoid confusion and to be sure to handle properly old arguments usage vs new arguments usage, I prefer keeping the old arguments (`branch` and `create_branch) and the new ones (`new_branch` and `commitish`). I will handle the conversion/compatibility check at the beginning of the operation.

My first API version was due to a bad understanding of the full `git worktree add` API (I used a small part of this API). Now, I think that the new API is more coherent with the `git worktree add` API :-)

> For `create_branch` - is it possible to create a worktree using an existing branch? Ie calling with `branch=x, create_branch=False`)?

Yes, it's possible. In this case, we have to use the `commititsh` argument which represents a git element from where to create the worktree (a commit, a branch, etc ...). So, for example:
```
        git.worktree(
            name='Create a worktree from the existing local branch `v1.0`',
            repo='/usr/local/src/pyinfra/master',
            worktree='/usr/local/src/pyinfra/hotfix',
            commitish='v1.0'
        )
```